dimensiondata/src/test/java/org/jclouds/dimensiondata/cloudcontrol/features/ServerApiLiveTest.java

```diff
@@ -0,0 +1,128 @@
+      assertNotNull(servers);
+      boolean foundDeployedServer = false;
+      for (Server s : servers) {
+         assertNotNull(s);
+         if (s.name().equals(deployedServerName)) {
+            foundDeployedServer = true;
+         }
+      }
+      assertTrue(foundDeployedServer, "Did not find deployed server " + deployedServerName);
+   }
+
+   @Test
+   public void testDeployAndStartServer() {
+      Boolean started = Boolean.TRUE;
+      NetworkInfo networkInfo = NetworkInfo
+            .create(NETWORK_DOMAIN_ID, NIC.builder().vlanId(VLAN_ID).build(), Lists.<NIC>newArrayList());
+      List<Disk> disks = ImmutableList.of(Disk.builder().scsiId(0).speed("STANDARD").build());
+      serverId = api().deployServer(deployedServerName, IMAGE_ID, started, networkInfo, "P$$ssWwrrdGoDd!", disks, null);
+      assertNotNull(serverId);
+      waitForServerStatus(api(), serverId, true, true, 30 * 60 * 1000, "Error");
+      waitForServerState(api(), serverId, State.NORMAL, 30 * 60 * 1000, "Error");
+   }
+
+   @Test(dependsOnMethods = "testDeployAndStartServer")
+   public void testReconfigureServer() {
+      api().reconfigureServer(serverId, 4, CpuSpeed.HIGHPERFORMANCE.name(), 1);
+      waitForServerState(api(), serverId, State.PENDING_CHANGE, 30 * 60 * 1000, "Error");
+      waitForServerState(api(), serverId, State.NORMAL, 30 * 60 * 1000, "Error");
+   }
+
+   @Test(dependsOnMethods = "testDeployAndStartServer")
+   public void testRebootServer() {
+      api().rebootServer(serverId);
+      waitForServerState(api(), serverId, State.PENDING_CHANGE, 30 * 60 * 1000, "Error");
+      waitForServerState(api(), serverId, State.NORMAL, 30 * 60 * 1000, "Error");
+      waitForVmToolsRunning(api(), serverId, 30 * 60 * 1000, "Error");
+   }
+
+   @Test(dependsOnMethods = "testRebootServer")
+   public void testPowerOffServer() {
+      api().powerOffServer(serverId);
+      waitForServerStatus(api(), serverId, false, true, 30 * 60 * 1000, "Error");
+   }
+
+   @Test(dependsOnMethods = "testPowerOffServer")
+   public void testStartServer() {
+      api().startServer(serverId);
+      waitForServerStatus(api(), serverId, true, true, 30 * 60 * 1000, "Error");
+      waitForVmToolsRunning(api(), serverId, 30 * 60 * 1000, "Error");
+   }
+
+   @Test(dependsOnMethods = "testStartServer")
+   public void testShutdownServer() {
+      api().shutdownServer(serverId);
+      waitForServerStatus(api(), serverId, false, true, 30 * 60 * 1000, "Error");
+   }
+
+   @Test(dependsOnMethods = "testShutdownServer")
+   public void testCloneServer() {
+      CloneServerOptions options = CloneServerOptions.builder().clusterId("").description("")
+            .guestOsCustomization(false).build();
+      cloneImageId = api().cloneServer(serverId, "ServerApiLiveTest-" + System.currentTimeMillis(), options);
+      assertNotNull(cloneImageId);
+      waitForServerState(api(), serverId, State.NORMAL, 30 * 60 * 1000, "Error");
+   }
+
+   @AfterClass
```

Comment:
Annotate with `alwaysRun = true` to let it run even if tests fail.

Reply:
Done